### PR TITLE
GEN-342 user should be able to mute certain tx

### PR DESCRIPTION
**Pull request by tal-blockfence**

### CODE REVIEW

**Comment by tal-blockfence** [comment]
Functionality enabled with no design yet.

Screenshot 2023-04-17 at 22 49 50

Screenshot 2023-04-17 at 22 43 35

cc @eyal-blockfence

File: src/components/UI/MuteButton.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+/* TODO Design Me */
+export function MuteButton({ address, chainId, url }: MuteButtonProps) {
+    const [mutedAddresses, setMutedAddresses] = usePersistentState<{ [key: string]: boolean }>('mutedAddresses', {});
+    const txId = txID(address, chainId, url);
+    const isMuted = mutedAddresses?.[txId];
+
+    async function toggleMute() {
+        setMutedAddresses({ ...mutedAddresses, [txId]: !isMuted });
```
Comment (tal-blockfence) [inline src/components/UI/MuteButton.tsx:19]: We should probably add a tooltip or conformation dialog

File: src/components/ButtonMenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import * as Menu from '../components/UI/Menu';
+import * as Styled from '../components/UI/Menu.styles';
+
+export interface MenuItemProps {
+    title: string;
+    body: string;
+    onClick: () => Promise<void>;
+}
+
+export function ButtonMenuItem({ title, body, onClick }: MenuItemProps) {
```
Comment (tal-blockfence) [inline src/components/ButtonMenuItem.tsx:11]: Design required..

File: src/components/UI/Menu.styles.tsx [hunk 3]
```diff
@@ -72,26 +72,29 @@ export const Item = styled.div`
...
+    user-select: none;
+    /* border-radius: 4px; */
 
     transition: 0.1s;
 
     opacity: 1;
 
     &:hover {
-        opacity: 0.8;
+        /* opacity: 0.8; */
```
Comment (tal-blockfence) [inline src/components/UI/Menu.styles.tsx:94]: nit: Clean up comment

File: src/components/UI/Menu.styles.tsx [hunk 4]
```diff
@@ -72,26 +72,29 @@ export const Item = styled.div`
...
+
     font-size: 0.8rem;
-    background-color: transparent;
     padding: 4px 16px;
-    color: black;
-    border: 1px solid rgba(255, 255, 255, 0.3);
-    border-radius: 4px;
+    width: 100%;
+    user-select: none;
+    /* border-radius: 4px; */
```
Comment (tal-blockfence) [inline src/components/UI/Menu.styles.tsx:87]: same

File: src/components/ContentDecoder.tsx [hunk 5]
```diff
@@ -65,6 +66,20 @@
                         <Risk key={id} risk={risk} />
                     ))}
                 </Collapsable>
+
+                {url && (
+                    <Styled.Options>
+                        <p>
+                            <b>Request Origin</b> <a href={url}>{url}</a>
```
[the remark this answers is not in the document]
Comment (tal-blockfence) [reply src/components/ContentDecoder.tsx:73]: @eyal-blockfence dismissing alert b/c the url is not user provided

File: src/components/ContentDecoder.tsx [hunk 6]
```diff
@@ -67,6 +66,20 @@ export function ContentDecoder({ to, chainId = '1', descriptionResult, analyzeRe
                         <Risk key={id} risk={risk} />
                     ))}
                 </Collapsable>
+
+                {url && (
```
Comment (tal-blockfence) [inline src/components/ContentDecoder.tsx:70]: Is there a better way to distinguish between wallet popup and manual search other than checking 'url' ?

File: src/components/UI/MuteButton.tsx (hunk 1 above)
Comment (tal-blockfence) [reply src/components/UI/MuteButton.tsx:22]: I see you skipped the tooltip, we can open a task to add it later

**Comment by tal-blockfence** [comment]
Adding Request change from Product:

We need to add a tooltip. In the first screen - by the end of the sentence near the checkbox, add a small i. It’s tooltip should be: “Enable this checkbox to prevent Blockfence pop-ups during transactions between this website and the designated blockchain address”
In the “Reset Muted Tx” - Change it to “Reset Mute Alerts”, and it’s tooltip should be: “Click this button to re-enable Blockfence pop-ups for transactions between any website and any blockchain address”

And the counter underneath should state: “Restore x muted alerts from select sites and addresses”

**Comment by tal-blockfence** [comment]
Final Design:

Screenshot 2023-04-20 at 16 13 50
Screenshot 2023-04-20 at 16 14 00

File: src/components/ContentDecoder.styles.tsx [hunk 7]
```diff
@@ -15,6 +15,20 @@ export const Subtitle = styled.h3`
     margin-top: 1.5rem;
 `;
 
+export const Options = styled.div`
+    /* border: 1px solid #dadce0; */
+    border-radius: 5px;
+
+    display: flex;
+    flex-direction: column;
+    /* padding: 8px 14px; */
```
Comment (gal-blockfence) [inline src/components/ContentDecoder.styles.tsx:24]:
```suggestion
    border-radius: 5px;

    display: flex;
    flex-direction: column;
```

File: src/manifest.json [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {
     "manifest_version": 3,
     "name": "Blockfence",
-    "version": "0.3.2",
+    "version": "0.0.0",
```
Comment (gal-blockfence) [inline src/manifest.json:4]: Is this change on purpose?

**Review by gal-blockfence — APPROVED** [review]
LGTM, just a couple of nits.
